docs/remote-k8s-access/README.md

```diff
@@ -0,0 +1,179 @@
+spec:
+  cassandra:
+    cluster: demo
+    serverVersion: "4.0.1"
+    storageConfig:
+      cassandraDataVolumeClaimSpec:
+        storageClassName: standard
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 5Gi
+    datacenters:
+      - metadata:
+          name: dc1
+        k8sContext: kind-k8ssandra-1
+        size: 3
+      - metadata:
+          name: dc2
+        k8sContext: kind-k8ssandra-2
+        size: 3
+      - metadata:
+          name: dc3
+        k8sContext: kind-k8ssandra-3
+        size: 3
+```
+
+Notice the `k8sContext` property of each element in the `datacenters` array. It should specify the name of a kube context that was provided by a ClientConfig. 
+
+The operator will use the value of `k8sContext` to look up the remote client from the cache and then use it to create the CassandraDatacenter in the remote cluster.
+
+As we can see, the ClientConfig is not used by the K8ssandraCluster. It only needs to know the kube context names.
+
+## Adding or removing a ClientConfig
+As stated earlier, the operator only processes ClientConfigs at startup. If you create or delete a ClientConfig after the operator has already started, it won't have any effect. You have to restart the operator for changes to take effect.
+
+Proceed with caution before deleting a ClientConfig. If there are any K8ssandraClusters that use the kube config provided by the ClientConfig, then the operator won't be able to properly manage them.
+
+## Creating a ClientConfig
+Creating a ClientConfig involves creating the kubeconfig file and secret. This can be error prone to do by hand. Instead use the `create-clientconfig.sh` script which can be found [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/scripts/create-clientconfig.sh).
+
+**Note:** The operator should already be installed in the remote cluster for which you want to create a ClientConfig.
+
+Here is a brief summary of what the script does:
+
+* Get the k8ssandra-operator service account from the data plane cluster
+* Extract the service account token
+* Extract the CA cert
+* Create a kubeonfig using the token and cert
+* Create a secret for the kubeconfig in the control plane custer
+* Create a ClientConfig in the control plane cluster that references the secret.
+
+Suppose we have two clusters with context name `kind-k8ssandra-0` and `kind-k8ssandra-1`. The control plane is running in `kind-k8ssandra-0`. We want to create a ClientConfig for `kind-k8ssandra-1`. This can be accomplished by running the following:
+
+```
+./create-clientconfig.sh --src-kubeconfig build/kubeconfigs/k8ssandra-1.yaml --dest-kubeconfig build/kubeconfigs/k8ssandra-0.yaml --in-cluster-kubeconfig build/kubeconfigs/updated/k8ssandra-1.yaml --output-dir clientconfig
+```
+The script stores all of the artifacts that it generates in a directory which is specified with the `--output-dir` option. If not specified, a temp directory is created.
```

Comment:
After creating the ClientConfig, should the user restart the operator?

Reply:
I added a note to remind the user to restart the operator.